File: linkedlist/linkedlist.js
class Node {
  constructor(data, next = null){
    this.data = data;
    this.next = next;
  }
}

function main(){
  const headNode = new Node(10, new Node(20, new Node(30, new Node(40, new Node(50, null)))));

  console.log("My Linked List: ");

  let first = "";
  let tempNode = headNode;

  while (tempNode !== null){
    first = first + tempNode.data + "\t";
    tempNode = tempNode.next;
  }
  console.log(first)
}

function averageOfElements(headNode){
  let count = 0;
  let sum = 0;

  while (headNode !== null){
    count++;
    sum += headNode.data;
    headNode = headNode.next;
  }

  const average = sum / count;
  return average;

}

main();
console.log("The average is: ", averageOfElements(headNode));

File: linkedlist/LinkedList.py
#!/usr/bin/env python3

class Node:
    def __init__ (self, data):
        self.data = data
        self.next = None
    def __init__ (self, data, next):
        self.data = data
        self.next = next

def main():

    headNode = Node(10, Node(20, Node(30, Node(40, Node(50, None)))))

    print("My linked list:")

    first = ''
    tempNode = headNode

    while tempNode != None:
        first = first+str(tempNode.data)+'\t'
        tempNode = tempNode.next

    print(first)

    print("The average of the elements: ", averageOfElements(headNode))

def averageOfElements(headNode):
    count = 0
    sum = 0

    while (headNode is not None):
        count += 1
        sum += headNode.data
        headNode = headNode.next

    average = sum / count
    return average

main()
